vglug/Hacktoberfest_Devops
# Implement Shell Scripts for File Counting, Renaming, and Extension-based Operations

Overview: This pull request adds four shell scripts that handle various file operations such as file counting and renaming. Each script automates specific tasks related to managing and renaming files based on extensions, prefixes, and user inputs.

Details of the Scripts:

Exercise 1:

Functionality: This script contains a file_count function that counts the number of regular files in the current working directory, excluding directories.
Purpose: Helps automate the task of quickly counting files.
Exercise 2:

Functionality: The file_count function is modified to accept a directory as an argument. It displays the directory name followed by the number of files within. The function is called for /etc, /var, and /usr/bin.
Purpose: Expands the file counting functionality to work with multiple directories.
Exercise 3:

Functionality: This script renames all .jpg files in the current directory by prepending the current date (YYYY-MM-DD) to the file names. If there are no .jpg files, it gracefully handles the case.
Purpose: Automates the process of renaming image files with a timestamp.
Exercise 4:

Functionality: This script prompts the user for a file extension and a prefix. It renames all files with the given extension by prepending the specified prefix. If no prefix is provided, the current date is used as the default.
Purpose: Allows flexible renaming of files based on file extension and user input.
Test Files: The PR also includes a script that generates test files for .jpg and other extensions to demonstrate and test the renaming functionalities in Exercises 3 and 4.

This pull request is part of my contributions for Hacktoberfest 2024 and aims to provide useful tools for file management automation using shell scripting.

How to Test:
Run the scripts in a bash-compatible shell.
Test with various file extensions and renaming cases.
Verify the renaming behavior with the provided test files.
